plugin/pkg/scheduler/factory/factory.go

```diff
@@ -1038,18 +1049,20 @@ func (factory *configFactory) MakeDefaultErrorFunc(backoff *util.PodBackoff, pod
 				Name:      pod.Name,
 			}
 
-			entry := backoff.GetEntry(podID)
-			if !entry.TryWait(backoff.MaxDuration()) {
-				glog.Warningf("Request for pod %v already in flight, abandoning", podID)
-				return
+			if !util.PodPriorityEnabled() {
+				entry := backoff.GetEntry(podID)
+				if !entry.TryWait(backoff.MaxDuration()) {
+					glog.Warningf("Request for pod %v already in flight, abandoning", podID)
+					return
+				}
 			}
 			// Get the pod again; it may have changed/been scheduled already.
 			getBackoff := initialGetBackoff
 			for {
 				pod, err := factory.client.CoreV1().Pods(podID.Namespace).Get(podID.Name, metav1.GetOptions{})
 				if err == nil {
 					if len(pod.Spec.NodeName) == 0 {
-						podQueue.AddIfNotPresent(pod)
+						podQueue.AddUnschedulableIfNotPresent(pod)
```

Comment:
This function is called when scheduler fails to schedule the pod. So, we place it in unschedulable queue, but unschedulable queue has more logic to decide whether it should go there or should go to active queue. It checks pod's status and whether a move request is received to decide between the two queues.

Reply:
OK, it seems probably safer to put it in the active queue, but if you're confident about this then it's fine with me.

Reply:
This is fine. If we put pods in the active queue here, then no pod will go to unschedulable queue.

Reply:
And in fact, putting the pod in active queue would cause a problem, because a very high priority pod which was not schedulable would block the scheduling queue (as it is now a priority queue).